File: code/community/1301690756020310020/implementing-language-detection-deepgram-api.js
const fs = require('fs');
const fetch = require('node-fetch');

const DEEPGRAM_API_KEY = process.env.DEEPGRAM_API_KEY;

async function transcribeAudio() {
  const audioData = fs.readFileSync('your_audio_file.mp3');

  const response = await fetch('https://api.deepgram.com/v1/listen', {
    method: 'POST',
    headers: {
      'Authorization': `Token ${DEEPGRAM_API_KEY}`,
      'Content-Type': 'audio/mpeg',
    },
    qs: {
      detect_language: 'true',
    },
    body: audioData
  });
  
  const data = await response.json();
  console.log(JSON.stringify(data, null, 2));
}

transcribeAudio().catch(console.error);

File: code/community/1301690756020310020/implementing-language-detection-deepgram-api.py
import os
import requests

# Set your Deepgram API key
DEEPGRAM_API_KEY = os.getenv('DEEPGRAM_API_KEY')

# Endpoint and headers
url = 'https://api.deepgram.com/v1/listen'
headers = {
    'Authorization': f'Token {DEEPGRAM_API_KEY}',
    'Content-Type': 'audio/mpeg',
}

params = {
    'detect_language': 'true',
}

# Reading audio file
with open('your_audio_file.mp3', 'rb') as audio_file:
    response = requests.post(url, headers=headers, params=params, data=audio_file)

print(response.json())